### PR TITLE
unit tests for the functions in the pkg/util/nicstobridge.go file

### PR DESCRIPTION
**Pull request by vjayaramrh**
**- What this PR does and why is it needed**

This PR is about adding unit tests for the functions in the pkg/util/nicstobridge.go file

**- Special notes for reviewers**

The files that are relevant to the current PR for review are:
- go-controller/pkg/util/nicstobridge.go (this file was modified to use the netLinkOps interface methods to make UT easy )
- go-controller/pkg/util/nicstobridge_test.go
- go-controller/pkg/util/ovs_unit_test.go (added a struct defintion to make easy mocking/adding string parameters)

**- How to verify it**

Pull this PR into your workspace
Navigate to go-controller/pkg/util directory
execute below commands
go test -coverprofile cp.out
go tool cover -html=cp.out
select file nicstobridge.go from drop down to verify coverage

**- Description for the changelog**

unit test coverage for functions in pkg/util/nicstobridge.go

### CODE REVIEW

File: go-controller/pkg/util/nicstobridge_test.go [hunk 1]
```diff
@@ -0,0 +1,1218 @@
...
+		{
+			desc:                "inpBrName WITHOUT prefix `br` and stdout empty",
+			outputEmpty:         true,
+			inpBrName:           "int",
+			onRetArgsExecIface:  &onCallReturnArgs{"RunCmd", []string{"*mocks.Cmd", "string", "[]string", "string", "string", "string", "string"}, []interface{}{bytes.NewBuffer([]byte("")), bytes.NewBuffer([]byte("")), nil}},
+			onRetArgsKexecIface: &onCallReturnArgs{"Command", []string{"string", "string", "string", "string", "string"}, []interface{}{mockCmd}},
+		},
+		{
+			desc:                "inpBrName WITH prefix `br` and stdout empty",
+			inpBrName:           "br-int",
```
Comment (trozet) [inline go-controller/pkg/util/nicstobridge_test.go:52]: where in this test do you actually check that the return value is "-int"?

Comment (vjayaramrh) [reply go-controller/pkg/util/nicstobridge_test.go:52]: The test is currently not comparing the actual output returned currently

Comment (vjayaramrh) [reply go-controller/pkg/util/nicstobridge_test.go:52]: I have now fixed the test case to compare output expected

File: go-controller/pkg/util/nicstobridge_test.go [hunk 2]
```diff
@@ -0,0 +1,1360 @@
...
+				{"RunCmd", []string{"*mocks.Cmd", "string", "[]string", "string", "string", "string", "string"}, []interface{}{bytes.NewBuffer([]byte("someoutput")), bytes.NewBuffer([]byte("")), nil}},
+			},
+			onRetArgsKexecIface: []onCallReturnArgs{
+				{"Command", []string{"string", "string", "string", "string", "string"}, []interface{}{mockCmd}},
+				{"Command", []string{"string", "string", "string", "string", "string", "string"}, []interface{}{mockCmd}},
+				{"Command", []string{"string", "string", "string", "string", "string", "string"}, []interface{}{mockCmd}},
+				{"Command", []string{"string", "string", "string", "string", "string"}, []interface{}{mockCmd}},
+			},
+		},
+		/*{
```
Comment (trozet) [inline go-controller/pkg/util/nicstobridge_test.go:144]: why is this commented out?

Comment (vjayaramrh) [reply go-controller/pkg/util/nicstobridge_test.go:144]: Oops, I need to remove the commented out tests, I have done a test refactor to accomodate the changes you made in an earlier PR that had changes to nictobridge.go
The commented ones need to be removed and I shall do it shortly and push an update to the PR

**Pull request merged**